kong/pdk/private/node.lua

```diff
@@ -0,0 +1,87 @@
+local utils = require "kong.tools.utils"
+local pl_file = require "pl.file"
+local pl_path = require "pl.path"
+local pl_dir = require "pl.dir"
+
+local ngx = ngx
+
+local cached_node_id
+
+local function node_id_filename(prefix)
+  return pl_path.join(prefix, "kong.id")
+end
+
+
+local function initialize_node_id(prefix)
+  if not pl_path.exists(prefix) then
+    local ok, err = pl_dir.makepath(prefix)
+    if not ok then
+      return false, "failed to create directory " .. prefix .. ": " .. err
```

Comment:
nit: it's convention to return `nil` on failure instead of `false`. The outcome is almost always the same, but it may confuse readers because an explicit `false` is typically used to signal some other kind of result.